connector/protobuf/src/test/scala/org/apache/spark/sql/protobuf/ProtobufSerdeSuite.scala

```diff
@@ -144,44 +146,52 @@ class ProtobufSerdeSuite extends SharedSparkSession with ProtobufTestBase {
   test("Fail to convert with missing Catalyst fields") {
     val protoFile = ProtobufUtils.buildDescriptor(testFileDesc, "FieldMissingInSQLRoot")
 
-    // serializing with extra fails if extra field is missing in SQL Schema
+    val foobarSQLType = DataType.fromDDL(
+      "struct<foo string>" // "bar" is missing.
```

Comment:
Shall we test catalyst schema which contains `bar` only and there is a result returned?

Reply:
Could you describe a bit more? 
Do we want to test not having 'foo' field which is optional?

Reply:
Yes

Reply:
Added a new test at line #69. PTAL.